vpr/src/pack/re_cluster_util.h

```diff
@@ -88,7 +88,8 @@ bool start_new_cluster_for_mol(t_pack_molecule* molecule,
                                t_clustering_data& clustering_data,
                                t_lb_router_data** router_data,
                                PartitionRegion& temp_cluster_pr,
-                               NocGroupId& temp_cluster_noc_grp_id);
+                               NocGroupId& temp_cluster_noc_grp_id,
+                               const int& detailed_routing_stage = 1);
```

Comment:
This new parameter needs to be commented here (add to the doxygen comment above).  Also comment why it defaults to 1.
Why is it being passed as a const int&?  An int is smaller than a reference, so it is simpler and slightly faster to just pass the int rather than a const int&.
@soheilshahrouz : I can also see that the temp_cluster_noc_grp_id is not commented in the function doxygen -- please add a comment for that in another PR.

Reply:
Is detailed_routing_stage on or off?  If so, a bool would be better.  If it can take a limited set of enumerated values, declaring it of the appropriate enumerated type is better. And if it can be arbitrary integers it needs to be carefully commented what that means.